instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -141,8 +151,69 @@ private static String stripSensitiveData(@Nullable String url) {
     }
 
     if (atIndex == -1 || atIndex == len - 1) {
-      return url;
+      return redactSensitiveParameters ? redactUrlParameters(url) : url;
+    }
+
+    String afterUserPwdRedaction = url.substring(atIndex);
+    return url.substring(0, schemeEndIndex + 3)
+        + "REDACTED:REDACTED"
+        + (redactSensitiveParameters
+            ? redactUrlParameters(afterUserPwdRedaction)
+            : afterUserPwdRedaction);
+  }
+
+  private static String redactUrlParameters(String urlpart) {
+
+    int questionMarkIndex = urlpart.indexOf('?');
+
+    if (questionMarkIndex == -1 || !containsParamToRedact(urlpart)) {
+      return urlpart;
+    }
+
+    boolean paramToRedact = false; // To be able to skip the characters of the parameters to redact
+    boolean paramNameDetected = false;
+    boolean reference = false;
+
+    StringBuilder urlPartAfterQuestionMark = new StringBuilder();
+
+    // To build a parameter name until we reach the '=' character
+    // If the parameter name is a one to redact, we will redact the value
+    StringBuilder currentParamName = new StringBuilder();
+
+    for (int i = questionMarkIndex + 1; i < urlpart.length(); i++) {
+      char currentChar = urlpart.charAt(i);
+      if (currentChar == '=') {
+        paramNameDetected = true;
+        urlPartAfterQuestionMark.append(currentParamName);
+        urlPartAfterQuestionMark.append('=');
+        if (PARAMS_TO_REDACT.contains(currentParamName.toString())) {
+          urlPartAfterQuestionMark.append("REDACTED");
+          paramToRedact = true;
+        }
+      } else if (currentChar == '&') { // New parameter delimiter
+        urlPartAfterQuestionMark.append('&');
+        paramNameDetected = false;
+        paramToRedact = false;
+        currentParamName.setLength(
+            0); // To avoid creating a new StringBuilder for each new parameter
+      } else if (currentChar == '#') { // Reference delimiter
+        reference = true;
+        urlPartAfterQuestionMark.append('#');
+      } else if (!paramNameDetected) {
+        currentParamName.append(currentChar);
+      } else if (!paramToRedact || reference) {
+        urlPartAfterQuestionMark.append(currentChar);
+      }
+    }
+    return urlpart.substring(0, questionMarkIndex) + "?" + urlPartAfterQuestionMark;
+  }
+
+  private static boolean containsParamToRedact(String urlpart) {
```

Comment:
did you consider that parameter names and values in url may be percent encoded?

Reply:
The parsing do processing based on the presence of the  `?`, `=`, `&` and `#` characters. All these characters are reserved ones after percent-encoding: https://en.wikipedia.org/wiki/Percent-encoding

Reply:
You have misunderstood, the problem isn't in these characters but rather in the names of the query parameters. For example `sig` could be encoded as `%73%69%67`. I'm not claiming that such encoding would be common or sensible, just that it is acceptable to the the web server.

Reply:
interesting, I suspect it's ok to ignore a query param encoded as `%73%69%67`, the goal of this masking isn't to account for malicious usage, but for valid usage of specific client libraries

@jeanbisutti can you open a semconv issue asking for this clarification?

Reply:
I also think it is fine to not bother with this since what is used by the http client is set by the application author. If needed we can always make the redacted parameter list configurable.

Reply:
https://github.com/open-telemetry/semantic-conventions/issues/1852

Reply:
thanks @jeanbisutti, I think we can proceed with the behavior in this PR (no percent encoding, exact case matching) and revisit after getting any semconv clarification